### PR TITLE
#102 Add support for nested expand and template rendering with expand

### PR DESCRIPTION
**Pull request by mitch2na**

### CODE REVIEW

File: json/src/main/groovy/grails/plugin/json/view/api/internal/DefaultJsonViewHelper.groovy [hunk 1]
```diff
@@ -3,10 +3,12 @@ package grails.plugin.json.view.api.internal
 import grails.plugin.json.builder.JsonGenerator
 import grails.plugin.json.builder.JsonOutput
 import grails.plugin.json.view.api.JsonView
+import grails.views.api.http.Parameters
 import grails.views.api.internal.DefaultGrailsViewHelper
 import grails.views.utils.ViewUtils
 import groovy.transform.CompileStatic
 import groovy.transform.InheritConstructors
+import org.apache.commons.collections.MapUtils
```
Comment (graemerocher) [inline json/src/main/groovy/grails/plugin/json/view/api/internal/DefaultJsonViewHelper.groovy]: Could you remove the dependency on commons collections.

File: json/src/main/groovy/grails/plugin/json/view/api/JsonView.groovy [hunk 2]
```diff
@@ -40,6 +40,17 @@ trait JsonView extends GrailsView {
      * The parent model, if any
      */
     Map parentModel
+
+    /**
+     * The nested view level used in serialization of expand parameter.
+     */
+    int level
```
Comment (graemerocher) [inline json/src/main/groovy/grails/plugin/json/view/api/JsonView.groovy]: Could you rename these something more specific to expand, as they have quite generic names

File: json/src/main/groovy/grails/plugin/json/view/api/JsonView.groovy [hunk 3]
```diff
@@ -40,6 +40,17 @@ trait JsonView extends GrailsView {
...
+
+    /**
+     * The first level arguments passed into the view.
+     */
+    Map arguments
```
Comment (graemerocher) [inline json/src/main/groovy/grails/plugin/json/view/api/JsonView.groovy]: Could you rename these something more specific to expand, as they have quite generic names

File: json/src/main/groovy/grails/plugin/json/view/api/internal/DefaultJsonViewHelper.groovy (hunk 1 above)
Comment (mitch2na) [inline json/src/main/groovy/grails/plugin/json/view/api/internal/DefaultJsonViewHelper.groovy]: Done

File: json/src/main/groovy/grails/plugin/json/view/api/JsonView.groovy (hunk 2 above)
Comment (mitch2na) [inline json/src/main/groovy/grails/plugin/json/view/api/JsonView.groovy]: Done named it with expand

**Pull request closed without merging**